From 5b595a393a98f84229cb7480fb8efdd5254edda3 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 18 Feb 2015 11:07:17 +0000
Subject: [PATCH 17/27] Arguments starting with '@@' are no paths

... so let's skip posix-to-windows conversion on such parameters.

This fixes Git's t1508-at-combinations.sh.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/msys2_path_conv.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/winsup/cygwin/msys2_path_conv.cc b/winsup/cygwin/msys2_path_conv.cc
index dd7bbd0..cfa0524 100644
--- a/winsup/cygwin/msys2_path_conv.cc
+++ b/winsup/cygwin/msys2_path_conv.cc
@@ -375,6 +375,11 @@ skip_p2w:
                     goto skip_p2w;
             }
             break;
+        case '@':
+            // Paths do not contain '@@'
+            if (it + 1 < end && it[1] == '@')
+                goto skip_p2w;
+        }
         ++it;
     }
     it = *src;
-- 
2.6.3

